go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/elastic/beats/v7
 
-go 1.22
+go 1.22.0
+
+toolchain go1.22.6
```

Comment:
I'd suggest to leave toolchain out of the go.mod. If you advocate to keep it, do we have to update any automation to bump this as well?

Reply:
We have to keep it out of `go.mod` otherwise  some of our mage targets will fail. IIRC some targets like `package` will mount some of the Go cache as read-only, but if `toolchain` is set, then the Go tool can try downloading that specific version and will fail with permission denied inside the docker image.

It's better not to set it to avoid those issues.
```suggestion
```

Reply:
@belimawr & @mauri870 `make check` complains if `toolchain go1.22.6` it isn't there because `go mod tidy` puts it in.  Any idea how to get around that behavior?

Reply:
> @belimawr & @mauri870 `make check` complains if `toolchain go1.22.6` it isn't there because `go mod tidy` puts it in. Any idea how to get around that behavior?

nevermind, figured it out.  collector was requiring 1.22.6, so I went back to a version of the collector that used go 1.21, and that seems works.  This should get better with repo work that Craig has planned.